### ISSUE TITLE
ERROR -- : run() in thread failed: uninitialized constant Listen::File::Digest

### ISSUE THREAD

**Issue by noahsw** [guard/listen#328]
I noticed Spring stopped noticing changes to my files so I began debugging Listen. This is what I found:

```
─± LISTEN_GEM_DEBUGGING=1 listen -v
I, [2015-07-08T14:49:55.922421 #18768]  INFO -- : Listen loglevel set to: 1
I, [2015-07-08T14:49:55.922496 #18768]  INFO -- : Listen version: 3.0.1
Starting listen...
I, [2015-07-08T14:49:56.387409 #18768]  INFO -- : Record.build(): 0.31961 seconds
> ["/Users/noah/Projects/scores-service/.idea/workspace.xml"]
> ["/Users/noah/Projects/scores-service/.idea/scores-service.iml", "/Users/noah/Projects/scores-service/.idea/workspace.xml"]
> ["/Users/noah/Projects/scores-service/app/models/app_logger.rb"]
> ["/Users/noah/Projects/scores-service/.idea/workspace.xml"]
W, [2015-07-08T14:50:24.956768 #18768]  WARN -- : scan DIED: uninitialized constant Listen::File::Digest:/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/file.rb:58:in `change'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/change.rb:58:in `invalidate'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/directory.rb:67:in `_change'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/directory.rb:26:in `block in scan'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/2.1.0/set.rb:263:in `each_key'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/2.1.0/set.rb:263:in `each'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/directory.rb:23:in `scan'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/change.rb:56:in `invalidate'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/base.rb:107:in `_queue_change'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:48:in `block in _process_event'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:43:in `each'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:43:in `_process_event'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/base.rb:42:in `block (2 levels) in configure'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rb-fsevent-0.9.5/lib/rb-fsevent/fsevent.rb:45:in `call'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rb-fsevent-0.9.5/lib/rb-fsevent/fsevent.rb:45:in `run'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:39:in `_run'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/base.rb:78:in `block in start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/internals/thread_pool.rb:6:in `call'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/internals/thread_pool.rb:6:in `block in add'
E, [2015-07-08T14:50:24.957823 #18768] ERROR -- : run() in thread failed: uninitialized constant Listen::File::Digest:\n /Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/file.rb:58:in `change'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/change.rb:58:in `invalidate'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/directory.rb:67:in `_change'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/directory.rb:26:in `block in scan'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/2.1.0/set.rb:263:in `each_key'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/2.1.0/set.rb:263:in `each'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/directory.rb:23:in `scan'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/change.rb:56:in `invalidate'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/base.rb:107:in `_queue_change'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:48:in `block in _process_event'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:43:in `each'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:43:in `_process_event'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/base.rb:42:in `block (2 levels) in configure'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rb-fsevent-0.9.5/lib/rb-fsevent/fsevent.rb:45:in `call'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/rb-fsevent-0.9.5/lib/rb-fsevent/fsevent.rb:45:in `run'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/darwin.rb:39:in `_run'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/adapter/base.rb:78:in `block in start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/internals/thread_pool.rb:6:in `call'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/internals/thread_pool.rb:6:in `block in add'\n\ncalled from:\n /Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/backend.rb:26:in `start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/listener.rb:67:in `block in <class:Listener>'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/fsm.rb:120:in `instance_eval'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/fsm.rb:120:in `call'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/fsm.rb:91:in `transition_with_callbacks!'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/fsm.rb:57:in `transition'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/listener.rb:90:in `start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/cli.rb:60:in `start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/lib/listen/cli.rb:30:in `start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
/Users/noah/.rbenv/versions/2.1.4/lib/ruby/gems/2.1.0/gems/listen-3.0.1/bin/listen:12:in `<top (required)>'
/Users/noah/.rbenv/versions/2.1.4/bin/listen:23:in `load'
/Users/noah/.rbenv/versions/2.1.4/bin/listen:23:in `<main>'
```

After this happens, Listen stops reporting all file changes.

**Comment by noahsw** [comment]
This is on Mac Yosemite

**Comment by noahsw** [comment]
More info:
1. Using `touch` does not trigger the error
2. Saving the file using Sublime does not trigger the error
3. Saving the file using RubyMine *does* trigger the error

**Comment by e2** [comment]
Thanks so much for reporting this - it can occurs only on OSX (for the most part). 

I'm not sure why the Travis tests didn't crash on OSX, but oh well...

I personally don't use OSX (Linux only), so I wouldn't have discovered this myself.

A fix and release is under way...

**Comment by e2** [comment]
Released as 3.0.2. Do let me know if something else is wrong.

Thanks!

**Comment by noahsw** [comment]
thanks for the quick turnaround! worked like a charm.
